src/sentry/integrations/github/client.py

```diff
@@ -264,13 +282,20 @@ def _populate_trees(self, repositories: List[Dict[str, str]]) -> Dict[str, RepoT
                     repo_info, only_use_cache, (3600 * 24) + (3600 * (index % 24))
                 )
             except ApiError as error:
-                self._populate_trees_process_error(error, extra)
+                should_count_error = self._populate_trees_process_error(error, extra)
+                if should_count_error:
+                    connection_error_count += 1
             except Exception:
                 # Report for investigation but do not stop processing
                 logger.exception(
                     "Failed to populate_tree. Investigate. Contining execution.", extra=extra
                 )
 
+            if connection_error_count >= MAX_CONNECTION_ERRORS:
+                raise TooManyApiErrors(
```

Comment:
Instead of raising an error, let's try to read from the cache with `only_use_cache`.

Reply:
Yup, that's a better idea, you're right. Done.